### PR TITLE
Update readme to include clone on build page

### PR DESCRIPTION
**Pull request by jonbgallant**
# Checklist
- [ x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-python/blob/master/.github/CONTRIBUTING.md).
- [ x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x ] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

Readme didn't include clone --recursive on build page.
# Description of the solution
 
Added clone

### CODE REVIEW

**Comment by jonbgallant** [comment]
@dominicbetts

File: doc/python-devbox-setup.md [hunk 1]
```diff
@@ -92,19 +109,25 @@ In order to setup your development environment to build the C binaries, you need
...
+1. Open a shell and navigate to the folder **build_all/mac** in your local copy of the repository.
+1. Run the `./setup.sh` script to install the prerequisite packages and the dependent libraries.
     * Setup will default to python 2.7
     * To setup dependencies for python version greater than 3, run `./setup.sh --python-version X.Y` where "X.Y" is the python version (e.g. 3.4, 3.5 or 3.6)
-4. Run the `./build.sh` script.
+1. Run the `./build.sh` script.
     * Build will default to python 2.7
     * To build with python version greater than 3, run `./build.sh --build-python X.X` where "X.Y" is the python version (e.g. 3.4, 3.5 or 3.6) 
-5. After a successful build, the `iothub_client.so` Python extension module is copied to the [**device/samples**][device-samples] and [**service/samples**][service-samples] folders. Visit these folders for instructions on how to run the samples.
+1. After a successful build, the `iothub_client.so` Python extension module is copied to the [**device/samples**][device-samples] and [**service/samples**][service-samples] folders. Visit these folders for instructions on how to run the samples.
```
Comment (zolvarga) [inline doc/python-devbox-setup.md]: Please fix the numbering here

File: doc/python-devbox-setup.md [hunk 2]
```diff
@@ -39,46 +39,63 @@ Because the Azure IoT SDKs for Python are wrappers on top of the [SDKs for C][az
...
 
-1.) On building the Python client library (`iothub_client.so`) on Linux devices that have less than **1GB** RAM, you may see build getting **stuck** at **98%** while building `iothub_client_python.cpp` as shown below
+1. When building the Python client library (`iothub_client.so`) on Linux devices that have less than **1GB** RAM, you may see build getting **stuck** at **98%** while building `iothub_client_python.cpp` as shown below
 
 ``[ 98%] Building CXX object python/src/CMakeFiles/iothub_client_python.dir/iothub_client_python.cpp.o``
 
 If you run into this issue, check the **memory consumption** of the device using `free -m command` in another terminal window during that time. If you are running out of memory while compiling iothub_client_python.cpp file, you may have to temporarily increase the **swap space** to get more available memory to successfully build the Python client side device SDK library.
 
-2.) CentOS7: Only Python 2.7 is supported due to a missing boost-python3 library package
+1. CentOS7: Only Python 2.7 is supported due to a missing boost-python3 library package
```
Comment (zolvarga) [inline doc/python-devbox-setup.md]: Please fix the numbering

File: doc/python-devbox-setup.md [hunk 3]
```diff
@@ -39,46 +39,63 @@ Because the Azure IoT SDKs for Python are wrappers on top of the [SDKs for C][az
...
+1. Open a shell and navigate to the folder **build_all/linux** in your local copy of the repository.
+1. Run the `./setup.sh` script to install the prerequisite packages and the dependent libraries.
     * Setup will default to python 2.7
     * To setup dependencies for python version greater than 3, run `./setup.sh --python-version X.Y` where "X.Y" is the python version (e.g. 3.4, 3.5 or 3.6)
-4. Run the `./build.sh` script.
+1. Run the `./build.sh` script.
     * Build will default to python 2.7
     * To build with python version greater than 3, run `./build.sh --build-python X.X` where "X.Y" is the python version (e.g. 3.4, 3.5 or 3.6)
-5. After a successful build, the `iothub_client.so` Python extension module is copied to the [**device/samples**][device-samples] and [**service/samples**][service-samples] folders. Visit these folders for instructions on how to run the samples.
+1. After a successful build, the `iothub_client.so` Python extension module is copied to the [**device/samples**][device-samples] and [**service/samples**][service-samples] folders. Visit these folders for instructions on how to run the samples.
```
Comment (zolvarga) [inline doc/python-devbox-setup.md]: Please fix the numbering

File: readme.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ Visit [Azure IoT Dev Center][iot-dev-center] to learn more about developing appl
 ## How to use the Azure IoT SDKs for Python
 Devices and data sources in an IoT solution can range from a simple network-connected sensor to a powerful, standalone computing device. Devices may have limited processing capability, memory, communication bandwidth, and communication protocol support. The IoT device SDKs enable you to implement client applications for a wide variety of devices.
 * **Using PyPI package on Windows, Linux (Ubuntu) or Raspberry Pi**: the simplest way to use the Azure IoT SDK for Python to develop device apps on Windows is to leverage the PyPI package which you can install following these [instructions][PyPI-install-instructions].
-* **Clone the repository**: The repository is using [GitHub Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) for its dependencies. In order to automatically clone these submodules, you need to use the --recursive option as described here:
+* **Clone the repository**: This repo uses [GitHub Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) for its dependencies. In order to automatically clone these submodules, you need to use the `--recursive` option as described here:
```
Comment (zolvarga) [inline readme.md]: >repo [](start = 33, length = 4)

"repository"

File: doc/python-devbox-setup.md (hunk 1 above)
Comment (jonbgallant) [reply doc/python-devbox-setup.md]: I updated, but so you know, markdown renderer will convert 
1.
1.

to 
1.
2.

**Comment by zolvarga** [comment]
Accepted and merged

**Pull request merged**